File: app/src/main/java/com/kode/utair/domain/repository/IWeatherSettingsRepository.java
package com.kode.utair.domain.repository;

import com.kode.utair.domain.models.WeatherSettings;

import io.reactivex.Completable;
import io.reactivex.Single;

public interface IWeatherSettingsRepository {

    Single<WeatherSettings> getWeatherSettings();

    Completable saveWeatherSettings(WeatherSettings weatherSettings);

}

File: app/src/main/java/com/kode/utair/domain/repository/IWeatherSettingsRepository.kt
package com.kode.utair.domain.repository

import com.kode.utair.domain.models.WeatherSettings
import io.reactivex.Completable
import io.reactivex.Single

interface IWeatherSettingsRepository {

    fun getWeatherSettings(): Single<WeatherSettings>

    fun saveWeatherSettings(weatherSettings: WeatherSettings): Completable

}